src/Uno.UWP/Media/Playback/MediaPlayer.wasm.cs

```diff
@@ -0,0 +1,297 @@
+﻿using System;
+using Uno.Media.Playback;
+using Windows.Media.Core;
+using Uno.Extensions;
+using System.IO;
+using Uno.Foundation.Logging;
+using System.Collections.Generic;
+using Uno;
+using Uno.Helpers;
+using System.Linq;
+
+namespace Windows.Media.Playback
+{
+	public partial class MediaPlayer
+	{
+		public IHtmlMediaPlayer Player;
+		private bool _isPlayRequested;
+		private bool _isPlayerPrepared;
+		private List<Uri> _playlistItems;
+		private int _playlistIndex;
+		const string MsAppXScheme = "ms-appx";
+
+		private void Initialize(IHtmlMediaPlayer player)
+		{
+			Player = player;
+			InitializePlayer();
+		}
+
+		private void TryDisposePlayer()
+		{
+			if (Player != null)
+			{
+				_isPlayRequested = false;
+				_isPlayerPrepared = false;
+			}
+		}
+
+		#region Player Initialization
+		private void InitializePlayer()
+		{
+			Player.OnSourceFailed -= OnError;
+			Player.OnSourceLoaded -= OnPrepared;
+			Player.OnSourceEnded -= OnCompletion;
+			Player.OnTimeUpdate -= OnTimeUpdate;
+			Player.OnSourceFailed += OnError;
+			Player.OnSourceLoaded += OnPrepared;
+			Player.OnSourceEnded += OnCompletion;
+			Player.OnTimeUpdate += OnTimeUpdate;
+
+			PlaybackSession.PlaybackStateChanged -= OnStatusChanged;
+			PlaybackSession.PlaybackStateChanged += OnStatusChanged;
+		}
+
+		private void SetPlaylistItems(MediaPlaybackList playlist)
+		{
+			_playlistItems = playlist.Items
+				.Select(i => i.Source.Uri)
+				.ToList();
+		}
+
+		protected virtual void InitializeSource()
+		{
+			PlaybackSession.NaturalDuration = TimeSpan.Zero;
+			PlaybackSession.PositionFromPlayer = TimeSpan.Zero;
+
+			// Reset player
+			TryDisposePlayer();
+			if (Source == null)
+			{
+				return;
+			}
+
+			try
+			{
+				InitializePlayer();
+				PlaybackSession.PlaybackState = MediaPlaybackState.Opening;
+
+				Uri uri;
+				switch (Source)
+				{
+					case MediaPlaybackList playlist when playlist.Items.Count > 0:
+						SetPlaylistItems(playlist);
+						uri = _playlistItems[0];
+						break;
+					case MediaPlaybackItem item:
+						uri = item.Source.Uri;
+						break;
+					case MediaSource source:
+						uri = source.Uri;
+						break;
+					default:
+						throw new InvalidOperationException("Unsupported media source type");
+				}
+
+				SetVideoSource(uri);
+				MediaOpened?.Invoke(this, null);
+			}
+			catch (global::System.Exception ex)
+			{
+				OnMediaFailed(ex);
+			}
+		}
+
+		private void OnTimeUpdate(object sender, object what)
+		{
+			PlaybackSession.PositionFromPlayer = Position;
+		}
+
+		private async void SetVideoSource(Uri uri)
```

Comment:
I definitely would avoid `async void`. I don't think fire and forget is a good idea.